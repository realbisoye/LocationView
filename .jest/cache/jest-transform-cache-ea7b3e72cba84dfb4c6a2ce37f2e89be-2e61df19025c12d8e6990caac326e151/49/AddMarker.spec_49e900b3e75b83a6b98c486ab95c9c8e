e1b427fc9dd77f207aa2b6b823507040
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");var _react = _interopRequireDefault(require("react"));
var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));
var _addMarker = _interopRequireDefault(require("../../app/components/addMarker"));var _jsxFileName = "/Users/bisoyeatolagbe/projects/bisoye/LocationView/__tests__/components/AddMarker.spec.js";

var marker = {
  longitude: -73.968285,
  latitude: 40.785091,
  description: 'Central Park playground update',
  title: 'Central Park' };


var updateMarker = jest.fn();
var addMarker = jest.fn();
var toggleForm = jest.fn();

describe('Add Marker', function () {
  var rendered = _reactTestRenderer.default.create(
  _react.default.createElement(_addMarker.default, {
    showForm: true,
    isUpdate: false,
    marker: {},
    updateMarker: updateMarker,
    toggleForm: toggleForm,
    addMarker: addMarker, __source: { fileName: _jsxFileName, lineNumber: 18 } }));



  it('renders Add New Marker Form', function () {
    expect(rendered).toMatchSnapshot();
  });

  it('set value on form value change', function () {
    rendered.root.instance.setValue('title', 'Menlo Park');
    expect(rendered.root.instance.state.marker.title).toEqual('Menlo Park');
  });

  it('set toggle form on request', function () {
    rendered.root.instance.closeForm();
    expect(toggleForm).toHaveBeenCalled();
  });

  it('create marker on submit', function () {
    rendered.root.instance.submit();
    expect(addMarker).toHaveBeenCalled();
  });

  it('renders Update New Marker Form', function () {
    var updateRendered = _reactTestRenderer.default.create(
    _react.default.createElement(_addMarker.default, {
      showForm: true,
      isUpdate: true,
      marker: marker,
      updateMarker: updateMarker,
      toggleForm: toggleForm,
      addMarker: addMarker, __source: { fileName: _jsxFileName, lineNumber: 49 } }));


    expect(updateRendered).toMatchSnapshot();
    expect(updateRendered.root.instance.state.marker).toEqual(marker);
  });

  it('update marker on submit for update', function () {
    var updateRendered = _reactTestRenderer.default.create(
    _react.default.createElement(_addMarker.default, {
      showForm: true,
      isUpdate: true,
      marker: marker,
      updateMarker: updateMarker,
      toggleForm: toggleForm,
      addMarker: addMarker, __source: { fileName: _jsxFileName, lineNumber: 64 } }));


    updateRendered.root.instance.submit();
    expect(updateMarker).toHaveBeenCalled();
  });
});