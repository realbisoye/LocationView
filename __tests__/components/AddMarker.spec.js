import React from 'react'
import renderer from 'react-test-renderer'
import AddMarker from '../../app/components/addMarker'

const marker = {
  longitude: -73.968285,
  latitude: 40.785091,
  description: 'Central Park playground update',
  title: 'Central Park',
}

const updateMarker = jest.fn()
const addMarker = jest.fn()
const toggleForm = jest.fn()

describe('Add Marker', () => {
  const rendered = renderer.create(
    <AddMarker
      showForm
      isUpdate={false}
      marker={{}}
      updateMarker={updateMarker}
      toggleForm={toggleForm}
      addMarker={addMarker}
    />,
  )

  it('renders Add New Marker Form', () => {
    expect(rendered).toMatchSnapshot()
  })

  it('set value on form value change', () => {
    rendered.root.instance.setValue('title', 'Menlo Park')
    expect(rendered.root.instance.state.marker.title).toEqual('Menlo Park')
  })

  it('set toggle form on request', () => {
    rendered.root.instance.closeForm()
    expect(toggleForm).toHaveBeenCalled()
  })

  it('create marker on submit', () => {
    rendered.root.instance.submit()
    expect(addMarker).toHaveBeenCalled()
  })

  it('renders Update New Marker Form', () => {
    const updateRendered = renderer.create(
      <AddMarker
        showForm
        isUpdate={true}
        marker={marker}
        updateMarker={updateMarker}
        toggleForm={toggleForm}
        addMarker={addMarker}
      />,
    )
    expect(updateRendered).toMatchSnapshot()
    expect(updateRendered.root.instance.state.marker.title).toEqual(marker.title)
    expect(updateRendered.root.instance.state.marker.description).toEqual(marker.description)
  })

  it('update marker on submit for update', () => {
    const updateRendered = renderer.create(
      <AddMarker
        showForm
        isUpdate={true}
        marker={marker}
        updateMarker={updateMarker}
        toggleForm={toggleForm}
        addMarker={addMarker}
      />,
    )
    updateRendered.root.instance.submit()
    expect(updateMarker).toHaveBeenCalled()
  })
})