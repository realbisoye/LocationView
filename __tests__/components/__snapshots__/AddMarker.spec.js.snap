// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Marker renders Add New Marker Form 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
>
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "#C2E098",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": "5%",
            "paddingHorizontal": 20,
            "top": 0,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#C2E098",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="menuButton"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#171914",
                "fontSize": 25,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
            "marginLeft": "22%",
            "zIndex": 1000,
          }
        }
        testID="apptitle"
      >
        Add Marker
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 20,
          "marginTop": 30,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#171914",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Title
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="words"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#88BF41",
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#171914",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Description
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#88BF41",
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#171914",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Latitude
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#88BF41",
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#171914",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Longitude
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#88BF41",
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <View
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "disabled",
            ]
          }
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <Text
              disabled={true}
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#BF41BA",
                  },
                  Object {
                    "color": "#cdcdcd",
                  },
                ]
              }
            >
              Add
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</Modal>
`;

exports[`Add Marker renders Update New Marker Form 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
>
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "#C2E098",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "maxHeight": "5%",
            "paddingHorizontal": 20,
            "top": 0,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#C2E098",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="menuButton"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#171914",
                "fontSize": 25,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
            "marginLeft": "22%",
            "zIndex": 1000,
          }
        }
        testID="apptitle"
      >
        Add Marker
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 20,
          "marginTop": 30,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#171914",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Title
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="words"
          defaultValue="Central Park"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#88BF41",
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value="Central Park"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#171914",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Description
        </Text>
        <TextInput
          allowFontScaling={true}
          defaultValue="Central Park playground update"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#88BF41",
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value="Central Park playground update"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#171914",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Latitude
        </Text>
        <TextInput
          allowFontScaling={true}
          defaultValue="40.785091"
          keyboardType="numeric"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#88BF41",
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value="40.785091"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#171914",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Longitude
        </Text>
        <TextInput
          allowFontScaling={true}
          defaultValue="-73.968285"
          keyboardType="numeric"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#88BF41",
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value="-73.968285"
        />
      </View>
      <View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              disabled={false}
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#BF41BA",
                  },
                ]
              }
            >
              Update
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</Modal>
`;
